source/contracts/libraries/token/BasicToken.sol

```diff
@@ -28,6 +28,18 @@ contract BasicToken is ERC20Basic {
     }
 
     /**
+    * @dev allows internal token transfers
+    * @param _from The source address
+    * @param _to The destination address 
+    */
+    function internalTransfer(address _from, address _to, uint256 _value) internal returns (bool) {
```

Comment:
The usage of this function should be applied throughout the codebase, as I suspect there are similar bugs elsewhere (and if there aren't there is opportunity for bugs elsewhere).  This could be done as a follow-up PR if you prefer that though.

Reply:
I'd prefer a follow up PR.